### PR TITLE
Provide a way to set smime signer on Message

### PR DESCRIPTION
**Pull request by ChadiRachid**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [X] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [X] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [X] It's submitted to the `develop` branch, _not_ the `master` branch if no hotfix
- [x] When resolving a specific issue, it's referenced in the PR's description (e.g. `Fix #xxx[,#xxx]`, where "xxx" is the Github issue number)
- [x] All tests are passing
- [ ] New/updated tests are included
- [ ] Changelog was modified

If adding a **new feature**, the PR's description includes:
- [X] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**
In some enterprise environments, all internal communication that targets a large number of recipients must be signed by a trusted certificate to meet company security policies.

Feature Request #3649

### CODE REVIEW

File: protected/humhub/components/mail/Mailer.php [hunk 1]
```diff
@@ -61,6 +71,9 @@ public function compose($view = null, array $params = [])
             $message->setFrom([Yii::$app->settings->get('mailer.systemEmailAddress') => Yii::$app->settings->get('mailer.systemEmailName')]);
         }
 
+        if($this->signingCertificatePath!=null && $this->signingPrivateKeyPath!=null)
```
Comment (kesselb) [inline protected/humhub/components/mail/Mailer.php]:
```suggestion
        if ($this->signingCertificatePath !== null && $this->signingPrivateKeyPath !== null)
```

File: protected/humhub/components/mail/Mailer.php [hunk 2]
```diff
@@ -61,6 +71,9 @@ public function compose($view = null, array $params = [])
             $message->setFrom([Yii::$app->settings->get('mailer.systemEmailAddress') => Yii::$app->settings->get('mailer.systemEmailName')]);
         }
 
+        if ($this->signingCertificatePath !== null && $this->signingPrivateKeyPath !== null)
+            $message->setSmimeSigner($this->signingCertificatePath, $this->signingPrivateKeyPath);
```
Comment (kesselb) [inline protected/humhub/components/mail/Mailer.php]: Please use `if (condition) { .... }`

**Comment by ChadiRachid** [comment]
> Looks good to me. Do you have some test plan? What kind of certificate is required?

**Do you have some test plan?**
No I do not have a test plan.

The actual Signing is handled by SwiftMailer, any tests regarding the handling of the certificate and the actual signing are dependent on the underlying library and are probably already done there.

The code suggested in this PR is passing Module parmeters defined in common.php to the swiftmailer library. Do we need a test for the _if !== null_  checks?

If it is recommended to have a test plan for the above, I will look into it, however I'm not entirely sure what is to be tested here.

**What kind of certificate is required?**
The type of certificate is an email signing/encryption certificate. Usually known as S/MIME Client certificate. 
The certificate and private key should be saved in PEM format and the path set in common.php
`        'mailer' => [
            'class' => 'humhub\components\mail\Mailer',
            'signingCertificatePath' => '/path/to/publiccert.pem',
            'signingPrivateKeyPath' => '/path/to/privatekey.pem'
        ],
`

File: protected/humhub/components/mail/Mailer.php (hunk 2 above)
Comment (ChadiRachid) [reply protected/humhub/components/mail/Mailer.php]: Made a commit for the code style https://github.com/humhub/humhub/pull/3651/commits/2c079be52a271a3c94a1e85eed9e64f2cdddbe6c

**Comment by ChadiRachid** [comment]
I need to add another enhancement to this specifically support for intermediate certificates and additional signing options. 

Should this be another issue/PR.  How would you recommend I proceed?

**Pull request merged**